### PR TITLE
Cross-target last few projects to DefaultNetCoreTargetFramework

### LINKED ISSUE #25917: Consider cross-compiling projects to target DefaultNetCoreTargetFramework
We have a handful of projects in the repo that do not cross-compile to .NET Core. The reference assemblies in .NET Core carry nullability and platform compatibility annotations and there are currently no plans to make this work on down level tfms.

Updating these projects to additionally cross-compile to `$(DefaultNetCoreTargetFramework)` seems like a small cost for the richer compile time we gain:

* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/clients/csharp/Client/src/Microsoft.AspNetCore.SignalR.Client.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/clients/csharp/Client.Core/src/Microsoft.AspNetCore.SignalR.Client.Core.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/clients/csharp/Http.Connections.Client/src/Microsoft.AspNetCore.Http.Connections.Client.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/common/Protocols.NewtonsoftJson/src/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson.csproj
*  https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/common/Protocols.MessagePack/src/Microsoft.AspNetCore.SignalR.Protocols.MessagePack.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/SignalR/server/StackExchangeRedis/src/Microsoft.AspNetCore.SignalR.StackExchangeRedis.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/Features/JsonPatch/src/Microsoft.AspNetCore.JsonPatch.csproj
* https://github.com/dotnet/aspnetcore/blob/ce16ff0a51a74811674228835696e5cc78494fd7/src/Logging.AzureAppServices/src/Microsoft.Extensions.Logging.AzureAppServices.csproj

### PR DESCRIPTION
**Pull request by wtgodbe**
Resolves https://github.com/dotnet/aspnetcore/issues/25917

### CODE REVIEW

**Comment by wtgodbe** [comment]
Actually there are a couple more, will update this

File: src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <Description>Distributed cache implementation of Microsoft.Extensions.Caching.Distributed.IDistributedCache using Microsoft SQL Server.</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (wtgodbe) [inline src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj:5]: @JunTaoLuo looks like this was from extensions - does the cross-targeting here make sense?

File: src/Caching/StackExchangeRedis/src/Microsoft.Extensions.Caching.StackExchangeRedis.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <Description>Distributed cache implementation of Microsoft.Extensions.Caching.Distributed.IDistributedCache using Redis.</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (wtgodbe) [inline src/Caching/StackExchangeRedis/src/Microsoft.Extensions.Caching.StackExchangeRedis.csproj:5]: @JunTaoLuo same question here

File: src/DataProtection/StackExchangeRedis/src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <Description>Support for storing data protection keys in Redis.</Description>
-    <TargetFrameworks>$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (wtgodbe) [inline src/DataProtection/StackExchangeRedis/src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.csproj:5]: @JunTaoLuo and here

File: src/Logging.AzureAppServices/src/Microsoft.Extensions.Logging.AzureAppServices.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <Description>Logger implementation to support Azure App Services 'Diagnostics logs' and 'Log stream' features.</Description>
-    <TargetFrameworks>$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (wtgodbe) [inline src/Logging.AzureAppServices/src/Microsoft.Extensions.Logging.AzureAppServices.csproj:5]: @JunTaoLuo and here

File: src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj (hunk 1 above)
Comment (JunTaoLuo) [reply src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj:5]: I don't see why not. Other extension packages like [Primitives](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Primitives/src/Microsoft.Extensions.Primitives.csproj#L3) are also cross-targeted.

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj [hunk 5]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <title>ASP.NET Core Logging Integration</title>
     <Description>This site extension enables logging integration for ASP.NET Core applications on Azure App Service.</Description>
-    <TargetFramework>$(DefaultNetFxTargetFramework)</TargetFramework>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (JunTaoLuo) [inline src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:6]: Does it make sense to cross-target site extensions? I thought these were built for specific target frameworks in a separate build. I could be wrong though so it's worth double checking. Same applies to LoggingBranch.

File: src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj (hunk 1 above)
Comment (dougbu) [reply src/Caching/SqlServer/src/Microsoft.Extensions.Caching.SqlServer.csproj:5]: Do it, do it, do it 😁

File: src/Caching/StackExchangeRedis/src/Microsoft.Extensions.Caching.StackExchangeRedis.csproj (hunk 2 above)
Comment (dougbu) [reply src/Caching/StackExchangeRedis/src/Microsoft.Extensions.Caching.StackExchangeRedis.csproj:5]: Same answer too

File: src/Logging.AzureAppServices/src/BlobLoggerProvider.cs [hunk 6]
```diff
@@ -77,7 +77,7 @@ internal override async Task WriteMessagesAsync(IEnumerable<LogMessage> messages
 
                     await writer.FlushAsync();
                     var tryGetBuffer = stream.TryGetBuffer(out var buffer);
-                    Debug.Assert(tryGetBuffer);
+                    System.Diagnostics.Debug.Assert(tryGetBuffer);
```
Comment (dougbu) [inline src/Logging.AzureAppServices/src/BlobLoggerProvider.cs:80]: There's a using for `System.Diagnostics`. Is `Debug` now ambiguous when building for `net6.0`❔ Don't do this if not.

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj (hunk 5 above)
Comment (dougbu) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:6]: We only build site extensions on Windows but that doesn't make this useless. @jkotalik or @BrennanConroy would `net6.0` site extension assemblies be useful❔

Comment (BrennanConroy) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:6]: Isn't the point of cross-targeting to get nicer behavior when consuming the package(s) (and sometimes perf benefits if you use newer APIs for different targets)?

Most use cases of the site extension are through the Azure Portal and so wouldn't care about the behavior consuming this package.

File: src/Logging.AzureAppServices/src/BlobLoggerProvider.cs (hunk 6 above)
Comment (wtgodbe) [reply src/Logging.AzureAppServices/src/BlobLoggerProvider.cs:80]: Without this we get:

> error CS0234: The type or namespace name 'Assert' does not exist in the namespace 'Microsoft.Extensions.Logging.Debug'

There are a couple other places in the repo where we do this, e.g. https://github.com/dotnet/aspnetcore/blob/7e145b1132ebe0bf85765f5ef96d337ff2d790e2/src/Http/Http/src/Features/FormFeature.cs#L262

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj (hunk 5 above)
Comment (wtgodbe) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: Reverted this change

File: src/SiteExtensions/LoggingBranch/LB.csproj [hunk 7]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <title>ASP.NET Core Extensions</title>
     <Description>This extension enables additional functionality for ASP.NET Core on Azure WebSites, such as enabling Azure logging.</Description>
-    <TargetFramework>$(DefaultNetFxTargetFramework)</TargetFramework>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework);$(DefaultNetFxTargetFramework);netstandard2.0</TargetFrameworks>
```
Comment (dougbu) [inline src/SiteExtensions/LoggingBranch/LB.csproj:6]: More site extensions
```suggestion
    <TargetFramework>$(DefaultNetFxTargetFramework)</TargetFramework>
```

**Review by dougbu — APPROVED** [review]
Once the remaining site extensions change is no more&hellip;

**Pull request merged**